Modules/Module_Processing/Module_ASL/xASL_wrp_RealignASL.m

```diff
@@ -525,14 +519,17 @@ function xASL_wrp_RealignASL(x, bASL)
         LoadParms = load(x.P.Path_ASL4D_mat, '-mat');
         mat = LoadParms.mat;
         
-        Incl = [];
-        for iExcl=1:length(exclusion)
-            if ~exclusion(iExcl)
-                Incl(end+1) = iExcl;
-            end
-        end
-        mat = mat(:,:,Incl);
+        mat = mat(:,:, ~exclusion);
+
         save(x.P.Path_despiked_ASL4D_mat, 'mat');
+
+        % Also change parameter vectors
+        x.Q.EchoTime = x.Q.EchoTime(~exclusion);

```

Comment:
RealignASL is called before subtractions?

Then it's quite possible that the 3 parameter vectors still don't have the full length. E.g. for most single-PLD sequences, we'll have at this stage only a single number. So we need to do the parameter replications to the full length of nVolumes using repmat...

Reply:
Yes, xASL_wrp_RealignASL is always called as second wrapper in xASL_module_ASL (after TopUp). Before all wrappers, at the start of xASL_module_ASL, we initialize the parameters (what you call parameter replications).

What we don't do here yet, which creates a bug, is storing these replicated parameters in the json sidecar. This only creates a bug currently for despiked_ASL4D (I added this to our follow-up issue)